test/Tests.Infrastructure/RavenTestBase.Cluster.cs

```diff
@@ -45,6 +46,37 @@ public async Task WaitForRaftCommandToBeAppliedInLocalServerAsync(string command
             await _parent.Server.ServerStore.Cluster.WaitForIndexNotification(updateIndex, TimeSpan.FromSeconds(10));
         }
 
+        public async Task<long> WaitForRaftCommandToBeAppendedInClusterAsync(List<RavenServer> nodes, string commandType, long timeout = 15_000, long interval = 100)
+        {
+            var sw = Stopwatch.StartNew();
+            long index = -1;
+            while (sw.ElapsedMilliseconds < timeout)
+            {
+                foreach (var server in nodes)
+                {
+                    try
+                    {
+                        var lastIndex = LastRaftIndexForCommand(server, commandType);
+                        index = lastIndex;
+                    }
+                    catch (TrueException)
```

Comment:
`TrueException` is a xunit exception, we don't usually catch those.
Also what if we waited for the 15sec we want to propogate the last exception (no matter what it was)
similar to `WaitForValueAsync`.
Also let's throw if we didn't find this on time

Reply:
Done